src/traces/bar/plot.js

```diff
@@ -37,8 +37,9 @@ module.exports = function plot(gd, plotinfo, cdbar) {
 
     var bartraces = plotinfo.plot.select('.barlayer')
         .selectAll('g.trace.bars')
-            .data(cdbar)
-      .enter().append('g')
+        .data(cdbar);
+
+    bartraces.enter().append('g')
```

Comment:
Yeah, I'd *love* to finish off the bar PR. It's 90% done, close, but the remaining 90% of the work required a moderate dive into the subtleties of bars.